src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -524,7 +524,7 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
             }
           }
         }
-
+        error_info.set_error_type(error_type);
```

Comment:
Since `RayErrorInfo` contains `ErrorType`, I don't understand why we are passing both around. But didn't refactor the arguments passing, but simply set the error type in the `RayErrorInfo` so that downstream functions could just take that.

Reply:
Is it correct though? It looks like it is set as WORKER_DIED even when worker is not died?

Reply:
Oh, when it's not correct? So it seems to me the logic here is if request worker lease failed, the default is `WORKER_DIED`, with all the above special cases? 

But I am not 100% if all the code above is correctly setting the error type and error info... But it does seem to me that `error_type` and `error_info` are being updated together.

Reply:
Ah, so this is not used if the lease succeeds right? That sgtm